Makefile

```diff
@@ -3,8 +3,11 @@ DLLINKFLAGS=-Wl,--kill-at -static-libgcc -shared
 
 all: edit-git-bash.dll
 
+# InnoSetup always uses the DLL target, and it must always be 32-bit
+# because InnoSetup is only 32-bit aware.
 edit-git-bash.dll: edit-git-bash.c
-	gcc $(CFLAGS) $(DLLINKFLAGS) -o $@ $^
+	PATH=/mingw32/bin:$$PATH \
+	i686-w64-mingw32-gcc -march=i686 $(CFLAGS) $(DLLINKFLAGS) -o $@ $^
```

Comment:
Let's test this. I have this 32-bit SDK lying around :smile: 

```sh
me@work MINGW32 /usr/src/build-extra
$ PATH=/mingw32/bin:$PATH i686-w64-mingw32-gcc
i686-w64-mingw32-gcc.exe: fatal error: no input files
compilation terminated.
```

Seems that it worked ;-)